sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateSchemaCompatibilityChecker.scala

```diff
@@ -66,7 +67,7 @@ class StateSchemaCompatibilityChecker(
     try {
       val versionStr = inStream.readUTF()
       // Ensure that version 3 format has schema file path provided explicitly
-      if (versionStr == "v3" && schemaFilePath.isEmpty) {
+      if (versionStr == "v3" && oldSchemaFilePath.isEmpty) {
```

Comment:
IIUC, is this supposed to be `newSchemaFilePath`?

Reply:
Ah, we don't actually need this. Will remove